### PR TITLE
Redesign: room directory makeover

### PR DESCRIPTION
**Pull request by bwindels**
https://github.com/vector-im/riot-web/issues/7607

A bit hacky in places, sorry.

roomdirectorymakeover

### CODE REVIEW

File: res/css/structures/_RoomDirectory.scss [hunk 1]
```diff
@@ -45,22 +64,17 @@ limitations under the License.
...
+    display: flex;
     margin-top: 12px;
     margin-bottom: 12px;
-    border-spacing: 5px;
 }
 
 .mx_RoomDirectory_searchbox {
-    display: table-cell;
-    vertical-align: middle;
+    flex: 1 !important;
```
Comment (turt2live) [inline res/css/structures/_RoomDirectory.scss]: do more specific styles help at all in this case? Something like `.mx_RoomDirectory_dialogWrapper .mx_RoomDirectory_searchBox` ?

File: src/i18n/strings/en_EN.json [hunk 2]
```diff
@@ -1497,5 +1497,6 @@
     "If you didn't remove the recovery method, an attacker may be trying to access your account. Change your account password and set a new recovery method immediately in Settings.": "If you didn't remove the recovery method, an attacker may be trying to access your account. Change your account password and set a new recovery method immediately in Settings.",
     "Failed to set direct chat tag": "Failed to set direct chat tag",
     "Failed to remove tag %(tagName)s from room": "Failed to remove tag %(tagName)s from room",
-    "Failed to add tag %(tagName)s to room": "Failed to add tag %(tagName)s to room"
+    "Failed to add tag %(tagName)s to room": "Failed to add tag %(tagName)s to room",
+    "Join": "Join"
```
Comment (turt2live) [inline src/i18n/strings/en_EN.json]: `npm run i18n` please to maintain ordering

File: res/css/structures/_RoomDirectory.scss (hunk 1 above)
Comment (bwindels) [reply res/css/structures/_RoomDirectory.scss]: Tried that, but the rule it's trying to override is very very specific (see `_dharma.scss:260`). The whole `mx_textinput`/ global `input` styles need to be better scoped anyways, so rather postpone this till then.

Comment (turt2live) [reply res/css/structures/_RoomDirectory.scss]: :(

+1 to postponing. It's not a major problem, just a thing we should avoid if possible

**Pull request merged**